website/addons/s3/static/s3NodeConfig.js

```diff
@@ -261,16 +272,78 @@ ViewModel.prototype.openCreateBucket = function() {
 
     var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d.-]{2,61}[^.]$/;
 
-    bootbox.prompt('Name your new bucket', function(bucketName) {
-        if (!bucketName) {
-            return;
-        } else if (isValidBucket.exec(bucketName) == null) {
-            bootbox.confirm({
-                title: 'Invalid bucket name',
-                message: 'Sorry, that\'s not a valid bucket name. Try another name?',
-                callback: function(result) {
-                    if (result) {
-                        self.openCreateBucket();
+    // Maps internal values used by API to display names
+    var BUCKET_LOCATION_MAP = $.extend(
+        {},
+        {
+            '': 'US Standard',
+            'EU': 'Europe Standard',
+            'us-west-1': 'California',
+            'us-west-2': 'Oregon',
+            'ap-northeast-1': 'Tokyo',
+            'ap-southeast-1': 'Singapore',
+            'ap-southeast-2': 'Sydney',
+            'cn-north-1': 'Beijing'
+        },
+        window.contextVars.s3Settings.bucketLocations
+    );
+
+    // Generates html options for key-value pairs in BUCKET_LOCATION_MAP
+    function generateBucketOptions(locations) {
+        var options;
+        for (var location in locations) {
+            if (BUCKET_LOCATION_MAP.hasOwnProperty(location)) {
+                options = options + ['<option value="', location, '">', locations[location], '</option>', '\n'].join('');
```

Comment:
Interesting. I'm not sure how it works exactly; maybe the the undefined is getting stripped out somewhere or maybe it's a browser quirk? 

<img width="480" alt="screen shot 2015-07-28 at 11 42 48 am" src="https://cloud.githubusercontent.com/assets/2614670/8936056/1aaf2a9a-351e-11e5-997c-cd028ef4b91a.png">

I'll go ahead and instantiate `options` as an empty string to be safe though